PD-2053 / 25.10 / PD-2053 Refresh SDLC content (by Mrt134) (#3915)
truenas/documentation: static/includes/TNLifecycle.md

The TrueNAS Software Development Life Cycle (SDLC) is the process of planning, creating, testing, deploying, and maintaining TrueNAS releases.

{{< tabs "SDLC Stages" >}}
{{< tab "Planning" >}}
Determine the objectives, nature, and scope of future versions of the software.
This involves gathering feedback and interpreting customer needs and requirements, diagnosing existing problems, and weighing the pros and cons of potential solutions.

> The end result is a list of recommended improvements to be integrated into a future TrueNAS versions.

Reviewer: ```suggestion
The end result is a list of recommended improvements to be integrated into a future TrueNAS version.
```

{{< /tab >}}
{{< /tabs >}}

This life cycle is generally applied to TrueNAS major versions released to the Community.
For Enterprise customers, releases are evaluated for Enterprise deployment suitability according to different criteria of internal testing on commercial hardware, security topics, upgradability, and general software stability as derived from Community usage.

A formal software End of Life (EoL) is not defined, as TrueNAS releases can be used in a stable manner for extended lengths of time, depending on use case and security risk factors.

> However, some general guidance is to evaluate the latest major version point release against your usage mindset.

Reviewer: ```suggestion
However, general guidance advises users to evaluate the latest major version point release against your usage mindset.
```
